### ISSUE TITLE
Following Readme I'm not able to install debos. (both docker and scratch installation)

### ISSUE THREAD

**Issue by r3v3rt3d** [go-debos/debos#236]
# From scratch installation cmd:

<pre>
$ sudo apt install golang git libglib2.0-dev libostree-dev qemu-system-x86 \
     qemu-user-static debootstrap systemd-container
$ mkdir -p /opt/src/gocode
$ export GOPATH=/opt/src/gocode
$ go get -u github.com/go-debos/debos/cmd/debos
</pre>

At the end using:
$ /opt/src/gocode/bin/debos: No such file or directory

#Docker installation:

<pre>
$ sudo docker pull godebos/debos
</pre>

Run docker image as suggested but i get the following error:

<pre>
$ sudo docker run --rm --interactive --tty --device /dev/kvm --user $(id -u) --workdir /recipes --mount "type=bind,source=$(pwd),destination=/recipes" --security-opt label=disable 52ed84763b21 example.yaml
Could not access KVM kernel module: Permission denied
qemu-system-x86_64: failed to initialize KVM: Permission denied
open /tmp/fakemachine-003177469/result: no such file or directory
</pre>

Can you hel me pls?

Regards.
Tommaso Merciai

**Comment by obbardc** [comment]
Hey,

You may need to add yourself to the kvm group.

Try something like (then reboot and retry the docker command):
sudo usermod -a -G $USER kvm

**Comment by Scott31393** [comment]
Hi,
Thanks for the tips and quick answer I figured out using:

<pre>
$ sudo chmod 777 /dev/kvm
</pre>

Can I create a pr for your readme?

Thanks in advance.

**Comment by obbardc** [comment]
Unfortunately after rebooting your box you won't be able to run debos since
the permissions of /dev/kvm goes back to normal.

Adding yourself to the kvm group really is the proper answer, that probably
should go in the readme and you're more than welcome to open a pr for that
;-)

**Comment by Scott31393** [comment]
Yes you are right but after use usermod cmd and restart I get the same problem :'(

<pre>
$ sudo usermod -a -G kvm $USER 
$ sudo docker run --rm --interactive --tty --device /dev/kvm --user $(id -u) --workdir /recipes --mount "type=bind,source=$(pwd),destination=/recipes" --security-opt label=disable 52ed84763b21 example.yaml
Could not access KVM kernel module: Permission denied
qemu-system-x86_64: failed to initialize KVM: Permission denied
open /tmp/fakemachine-655363751/result: no such file or directory
</pre>

Maybe is my host configuration.

**Comment by obbardc** [comment]
Can you please post the output of

$ groups

$ ls -la /dev/kvm

**Comment by Scott31393** [comment]
$ ls -la /dev/kvm 
crw-rw----+ 1 root kvm 10, 232 ott 27 00:20 /dev/kvm

**Comment by obbardc** [comment]
so after a reboot it should work, since you're in the kvm group ;-)

**Comment by Scott31393** [comment]
Thanks again for the tips. Another question. How can I get debos binary and run it? I follow your readme but can't find bin folder. Today I test this fs on imx8 and imx6 SOC.


Thanks in advance.

Tommaso

**Comment by obbardc** [comment]
Can you try the following and post the output? Maybe you don't have the right permissions for /opt
```
$ export GOPATH=~/go
$ mkdir -p $GOPATH
$ go get -u github.com/go-debos/debos/cmd/debos
$ /opt/src/gocode/bin/debos --help
```

**Comment by Scott31393** [comment]
I follow your tips:

```
$ export GOPATH=~/go
$ mkdir -p $GOPATH
$ go get -u github.com/go-debos/debos/cmd/debos
package math/bits: unrecognized import path "math/bits" (import path does not begin with hostname)
$ ls /home/tom/go/src/gocode
ls: cannot access '/home/tom/go/src/gocode': No such file or directory
```


Another question:
In which way I can compile for 32bit arm architecture instead of arm64?

Inside .yaml I change the following line:
```
architecture: arm64 --> architecture: arm
```

```
Running /debos --artifactdir /recipes /recipes/example.yaml
2020/10/27 08:28:36 ==== debootstrap ====
2020/10/27 08:28:36 Debootstrap | W: Unable to read /etc/apt/apt.conf.d/ - DirectoryExists (2: No such file or directory)
2020/10/27 08:28:36 Debootstrap | I: Retrieving InRelease 
2020/10/27 08:28:36 Debootstrap | I: Checking Release signature
2020/10/27 08:28:36 Debootstrap | I: Valid Release signature (key id 6D33866EDD8FFA41C0143AEDDCC9EFBF77E11517)
2020/10/27 08:28:37 Debootstrap | E: Invalid Release file, no entry for main/binary-arm/Packages
2020/10/27 08:28:37 debootstrap.log | 2020-10-27 08:28:36 URL:https://deb.debian.org/debian/dists/buster/InRelease [121480/121480] -> "/scratch/root/var/lib/apt/lists/partial/deb.debian.org_debian_dists_buster_InRelease" [1]
2020/10/27 08:28:37 debootstrap.log | gpgv: Signature made Sat Sep 26 09:56:29 2020 UTC
2020/10/27 08:28:37 debootstrap.log | gpgv:                using RSA key 16E90B3FDF65EDE3AA7F323C04EE7237B7D453EC
2020/10/27 08:28:37 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>"
2020/10/27 08:28:37 debootstrap.log | gpgv: Signature made Sat Sep 26 09:56:30 2020 UTC
2020/10/27 08:28:37 debootstrap.log | gpgv:                using RSA key 0146DC6D4A0B2914BDED34DB648ACFD622F3D138
2020/10/27 08:28:37 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (10/buster) <ftpmaster@debian.org>"
2020/10/27 08:28:37 debootstrap.log | gpgv: Signature made Sat Sep 26 10:02:41 2020 UTC
2020/10/27 08:28:37 debootstrap.log | gpgv:                using RSA key 6D33866EDD8FFA41C0143AEDDCC9EFBF77E11517
2020/10/27 08:28:37 debootstrap.log | gpgv: Good signature from "Debian Stable Release Key (10/buster) <debian-release@lists.debian.org>"
2020/10/27 08:28:37 Action `debootstrap` failed at stage Run, error: exit status 1
Powering off.
```
But this give me some error :)

Can you help me?

Thanks again for support.

**Comment by obbardc** [comment]
> I follow your tips:
> 
> ```
> $ export GOPATH=~/go
> $ mkdir -p $GOPATH
> $ go get -u github.com/go-debos/debos/cmd/debos
> package math/bits: unrecognized import path "math/bits" (import path does not begin with hostname)
> $ ls /home/tom/go/src/gocode
> ls: cannot access '/home/tom/go/src/gocode': No such file or directory
> ```

Looks like you need a newer version of Go, can you post the output of `go version`?
For reference I am using `go version go1.15.2 linux/amd64`

> 
> Another question:
> In which way I can compile for 32bit arm architecture instead of arm64?
> 
> Inside .yaml I change the following line:
> 
> ```
> architecture: arm64 --> architecture: arm
> ```

try `armhf` or `armel` (i think it would be armhf). See here for more info ;-) https://www.debian.org/ports/arm/

**Comment by Scott31393** [comment]
Thanks. Maybe I need to update :)
$ go version
go version go1.6.2 linux/amd64

I let you know.

**Comment by Scott31393** [comment]
Update to:

```
$ go version
go version go1.15.3 linux/amd64
```

But have the same issue. :'(

**Comment by nlseven** [comment]
For /dev/kvm permission issue above: keep in mind that, depending on how you run it, the kvm group _inside_ the container may have a different GID from the host.
You could try running it with `--group-add` and the actual GID that KVM is expecting:

```bash
docker run --rm --interactive --tty --device /dev/kvm --user $(id -u) --group-add $(stat -c '%g' /dev/kvm) --workdir /recipes --mount "type=bind,source=$(pwd),destination=/recipes" --security-opt label=disable godebos/debos -b kvm example.yaml
```

**Comment by wailashi** [comment]
The following command works with rootless Podman on Debian 12:

```
podman run --rm --interactive --tty -v /dev/kvm:/dev/kvm --workdir /recipes --mount "type=bind,source=$(pwd),destination=/recipes" --tmpfs /dev/shm:rw,size=256M,mode=1777 --group-add keep-groups godebos/debos -b kvm example.yaml
```

The user must be in the `kvm` group. If you encounter segfaults, increasing the size of the `shm` mount might help.
